### ISSUE TITLE
Unexpected warning CS0034: Operator '-' is ambiguous on operand of type 'nuint' and 'int' 

### ISSUE THREAD

**Issue by buyaa-n** [dotnet/roslyn#63860]
When adding or subtracting an int variable to / from  UIntPtr type cause ambiguous operator warning, this is not happening for IntPtr type or adding/substructing literals

**Steps to Reproduce**:
```cs
class Program
{
    IntPtr intPtr1;
    IntPtr intPtr2;
    UIntPtr uintPtr1;
    UIntPtr uintPtr2;

    void M1()
    {
        int i = 1;
        uintPtr2 = uintPtr1 + 2;
        uintPtr2 = uintPtr1 - i; // warning CS0034: Operator '-' is ambiguous on operand of type 'nuint' and 'int'
        intPtr1 = intPtr2 - i; // no warning
    }
}
```

**Expected Behavior**:
Now warning

**Actual Behavior**:
Warns for UIntPtr

**Comment by jcouv** [comment]
Thanks for reporting this. This break is an expected change in C# 11.
I've added documentation, including some workarounds at https://github.com/dotnet/roslyn/blob/main/docs/compilers/CSharp/Compiler%20Breaking%20Changes%20-%20DotNet%207.md#addition-of-systemuintptr-and-systemint32
